interpreter/for_test.go

```diff
@@ -546,6 +547,7 @@ func TestInterpretEphemeralReferencesInForLoop(t *testing.T) {
 	t.Run("Mutating reference to resource array", func(t *testing.T) {
 		t.Parallel()
 
+		// TODO: Use compiler (need mutation-while-iterating validation)
```

Comment:
Mutation while iteration is not straightforward to implement. Hence left this for now, and added as an item for https://github.com/onflow/cadence/issues/3804